CRM/Price/BAO/LineItem.php

```diff
@@ -834,7 +834,8 @@ protected function getAdjustedFinancialItemsToRecord($entityID, $entityTable, $c
       // INSERT a financial item to record surplus/lesser amount when a text price fee is changed
       elseif (!empty($lineItemsToUpdate) &&
       $lineItemsToUpdate[$updateFinancialItemInfoValues['price_field_value_id']]['html_type'] == 'Text' &&
-      $updateFinancialItemInfoValues['amount'] > 0
+      $updateFinancialItemInfoValues['amount'] > 0 &&
+      ($lineItemsToUpdate[$updateFinancialItemInfoValues['price_field_value_id']]['line_total'] - $totalFinancialAmount) > 0
```

Comment:
This is to ensure that for the unchanged text price fee no new financial item (amount = $0) will be created.